### PR TITLE
add tests and refactor to meet python syntax format guidelines

### PR DESCRIPTION
**Pull request by ebarnsli**
So I did quite a few things to your code. Mocked some of your stuff so i was able to create base tests for multiChoice and singleChoice... really boosted your test coverage. 

I made some code changes. most of these were best practices or standards for python development :) 

- I removed all redundant parenthesis, this is a code format suggestion that's standard for python. 
- there was a lot of redundant return values so we got rid of those
- made some private methods of small tasks that can be given a method name instead of a consistent comment, like getting input
- move tests to parameterized tests. This can save a lot of space on the page when testing a lot of inputs 
- added tests for testing printed output of cursor methods

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,209 +1,230 @@
...
+        return 0
     return -1
+
+
 def cursorRight(val):
-    if(isinstance(val, int)):
-        if((val)>0):
-            print(u"\u001b["+str(val)+"C",end='')
-            return 0
-        return -1
```
Comment (ebarnsli) [inline main.py:22]: these return -1 are redundant

File: main.py [hunk 2]
```diff
@@ -1,209 +1,230 @@
...
+                  "GREEN": "\u001b[32m",
+                  "YELLOW": "\u001b[33m",
+                  "BLUE": "\u001b[34m",
+                  "MAGENTA": "\u001b[35m",
+                  "CYAN": "\u001b[36m",
+                  "WHITE": "\u001b[37m",
+                  "RESET": "\u001b[0m"}
+        if val in colors.keys():
+            print(colors[val], end="")
+            return 0
```
Comment (ebarnsli) [inline main.py:51]: reads a little nicer

File: main.py [hunk 3]
```diff
@@ -1,209 +1,230 @@
...
         sys.stdout.flush()
-        charIn=ord(sys.stdin.read(1))#Get input
-        if(charIn==27):#ESC pressed
-            charIn=ord(sys.stdin.read(1))
-            if(charIn==91):#[ pressed
-                charIn=ord(sys.stdin.read(1))#Get input
-                if(charIn==65):#Up
-                    if(position>0):
-                        print(" ",end="")
+        charIn = get_input_char()  # Get input
```
Comment (ebarnsli) [inline main.py:87]: turn ord(sys.stdin.read(1)) into a function so you know what it does implicitly and don't need a comment

File: main.py [hunk 4]
```diff
@@ -1,209 +1,230 @@
...
-    cursorDown(len(arr)-position)
-    print("",end='\n\r')
-    #Set terminal style back to normal
+
+        elif charIn == 27:  # ESC pressed
+            charIn = ord(sys.stdin.read(1))
+            if charIn == 91:  # [ pressed
+                charIn = ord(sys.stdin.read(1))  # Get input
+                if charIn == 65:  # Up
+                    position = onCharUp(cursorColor, position)
```
Comment (ebarnsli) [inline main.py:189]: broke onCharUp and onCharDown into private functions, helps readability to not see blocks of 10 commands

File: test.py [hunk 5]
```diff
@@ -1,73 +1,70 @@
 import pytest
 from main import *
 
-def test_Tester(capfd):
-    print("Hello World!")
+
+@pytest.mark.parametrize("test_input,expected",
+                         [(0, -1), (-1, -1), (-100, -1),
+                          (0.1, -1), ("N", -1), ('f', -1), ([1, 2], -1), (1, 0), (5, 0)])
```
Comment (ebarnsli) [inline test.py:9]: first value in tuple is input, second is expected Input: 0 expected: -1

**Review by Hammie217 — APPROVED** [review]
Looks great, thanks

**Pull request merged**